src/printer-markdown.js

```diff
@@ -133,24 +135,121 @@ function genericPrint(path, options, print) {
     }
     case "thematicBreak":
       return concat(["---", hardline]);
+    case "table":
+      return printTable(path, options, print);
+    case "tableCell":
+      return printChildren(path, options, print);
+    case "tableRow": // handled in "table"
     default:
       throw new Error(`Unknown markdown type ${JSON.stringify(node.type)}`);
   }
 }
 
-function hasParentType(path, type) {
+function hasParentType(path, typeOrTypes) {
+  const types = [].concat(typeOrTypes);
+
   let counter = 0;
   let parentNode;
 
   while ((parentNode = path.getParentNode(counter++))) {
-    if (parentNode.type === type) {
+    if (types.indexOf(parentNode.type) !== -1) {
       return true;
     }
   }
 
   return false;
 }
 
+function printTable(path, options, print) {
+  const node = path.getValue();
+  const contents = []; // { [rowIndex: number]: { [columnIndex: number]: string } }
+
+  path.map(rowPath => {
+    const rowContents = [];
+
+    rowPath.map(cellPath => {
+      rowContents.push(
+        printDocToString(cellPath.call(print), options).formatted
+      );
+    }, "children");
+
+    contents.push(rowContents);
+  }, "children");
+
+  const columnMaxWidths = contents.reduce(
```

Comment:
We should also limit the max length of the columns. For example, All-Contributors tables contain HTML code and table dash columns processed by Prettier get very very long.

Reply:
Can you open a new issue so we can track this better?

Reply:
Done https://github.com/prettier/prettier/issues/4195